packages/emotion/src/styleUtils/polyFill.js

```diff
@@ -152,7 +152,7 @@ const processStyleProps = (propsObj, dir) =>
             }
           }
 
-          if (isSupportedProps(originalProp) && originalValue) {
+          if (isSupportedProps(originalProp) && originalValue !== 'undefined') {
```

Comment:
@Brailor had to make a small tweak, because it was ignoring props with `0` (as number) values, for example `insetInlineStart: 0`.